covid/question_covid.py

```diff
@@ -67,10 +65,19 @@ def predict(self, question):
             with open(paper_path) as json_file:
                 article_data = json.load(json_file)
                 text = ' '.join([d['text'] for d in article_data['body_text']])
-            sentences = text.split('.')
-            n = 3
-            sentences_grouped = ['.'.join(sentences[i:i+n]) for i in range(0, len(sentences), n)]
-            for subtext in sentences_grouped:
+            sentences = [s.text for s in self.scispacy(text).sents]
+            
+            def yield_subtext(sentences):
+                subtext = ''
+                for i in range(len(sentences)):
+                    sent = sentences[i]
+                    if len(sent) + len(subtext) > 450:
+                        yield subtext
+                        subtext = sent
```

Comment:
I don't think this works if `len(sent)>450`? 
e.g. 
`subtext =''`
`sent = 'kdfjhsdfkjhsdfjk ....skdjfhdkfjhdkfh'` (len>450)
`len(sent) + len(subtext) > 450` is true
`yield ''`
`subtext = sent`
the next iteration we will yield subtext which is `'kdfjhsdfkjhsdfjk ....skdjfhdkfjhdkfh'` (len>450)

I think in this (hopefully rare) situation we will just have to chop up subtext

Reply:
there are some edge cases indeed, will try to fix today but merge first